python/ray/util/sgd/torch/training_operator.py

```diff
@@ -79,6 +92,20 @@ def __init__(self,
         self.timers = TimerCollection()
         self.setup(config)
 
+    def _set_batch_logs_reporter(self, r):
+        self._batch_logs_reporter = r
+
+    def _send_setup_info(self, data):
+        return ray.get(self._batch_logs_reporter._send_setup.remote(data))
+
+    def _send_batch_logs(self, data):
+        cur_time = time.monotonic()
+        if cur_time - self._last_batch_logs_send < BATCH_LOGS_RATE_LIMIT:
+            return
+
+        self._last_batch_logs_send = cur_time
+        return ray.get(self._batch_logs_reporter._send.remote(data))
+
```

Comment:
As mentioned offline, I was thinking that all this logic be hidden in a separate handler or class. 

Maybe one good rule of thumb is for the operator not to need to call Ray primitives (not sure)